java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -920,4 +1042,34 @@ private boolean endsWithReturn(StatementTree stmt) {
     }
     return false;
   }
+
+  /**
+   * Utility method. Checks whether the value associated to a given map and given key is a non-empty
+   * string.
+   *
+   * @param map - map corresponding to a method property
+   * @param key - key to check the corresponding value
+   * @return true if value is a non-empty string
+   */
+  private boolean checkValueStringExistsAndNotEmpty(Map<String, Object> map, String key) {
+    Object value = map.get(key);
+    return value instanceof String && !value.equals(EMPTY);
+  }
+
+  /**
+   * Utility method. Checks whether the argumentIndex key of a method property map is a non-negative
+   * integer.
+   *
+   * @param map - map corresponding to a method property
+   * @return true if argument index is a non-negative integer
+   */
+  private boolean checkArgumentIndexExistsAndIsValid(Map<String, Object> map) {
+    Object value = map.get(ARGUMENT_INDEX_STRING);
+    if (value instanceof Long) {
+      return ((Long) value).intValue() >= ZERO;
+    } else if (value instanceof Integer) {
+      return ((Integer) value) >= ZERO;
```

Comment:
Still don't see the need for checking `Integer`.

Reply:
Might be required if different parser is used eg. when config support is added for YAML.
Removing it for now.